CodeU_Assignement_3/src/codeu_assignement_3/SimpleDictionary.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package codeu_assignement_3;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import static javafx.scene.input.KeyCode.T;
+
+/**
+ *
+ * @author Karine
+ * Simple dictionary only for testing
+ * 
+ * Consider using mocked dictionary instead of this implementation
+ * 
+ */
+public class SimpleDictionary implements KemDictionary 
+{
+    /* Dictionary as an array */
+    private final ArrayList<String> m_dict = new ArrayList<>(); 
+    
+    
+    /* Add all in BFS order (i.e., Breadth-First Traversal of a Tree) */
+    public SimpleDictionary(String[] wordsInDict)
+    {
+        if (wordsInDict != null)
+        {
+            List<String> temp = Arrays.asList(wordsInDict);
+            temp.replaceAll(String::toUpperCase);
+            m_dict.addAll(temp);
+        }
+    }
+    
+    /* 
+       Returns whether the given string is a valid word 
+    */
+    @Override
+    public boolean isWord (String w)
+    {
+        return m_dict.contains(w.toUpperCase());
+    }
+    
+    
+    /* 
+       Returns whether the given string is a prefix of at 
+       least one word in thedictionary 
+    */
+    @Override
+    public boolean isPrefix (String p)
+    {
+        if (isWord(p)) 
+        {
+            return true;
+        }

```

Comment:
This is the wrong approach:
- This is supposed to be a simple and quick implementation of the Dictionary interface. Don't add unnecessary complexity.
- If you want it to be efficient, the correct solution is to make the main prefix search fast, not add a special case onto it.
- Even if you couldn't make the prefix search faster, this is an optimisation of marginal use. It accelerates the leaves of the search tree, ignoring that most isPrefix calls are not looking up complete words. Making the occasional complete word look-up faster will not stop the main prefix check loop from dominating the cost.
- The isWord check has the same algorithmic complexity as the prefix loop - m_dict is a list, so isWord is O(n) in the dictionary size anyway, just like the prefix-checking loop.

There's a popular saying "Premature optimisation is the root of all evil.". It's important to know when not to optimise, as well as when to optimise, and this is one of those cases.